### PR TITLE
Fixed --keep_invisible defaults, closes #3514

### LINKED ISSUE #3514: Lossy cjxl with Alpha channel not using --keep_invisible=0 as default
**Describe the bug**
When using `cjxl` to encode an image with the Alpha channel to a lossy JPEG XL file, the default value for `keep_invisible` should be 0. The results show a possible default as 1 instead.

```
--keep_invisible=0|1
disable/enable preserving color of invisible pixels (default: 1 if lossless, 0 if lossy).
```

**To Reproduce**
```
cjxl file.png file.jxl -d 1
cjxl file.png file.jxl -d 1 --keep_invisible=0
```

**Expected behavior**
The default value of `keep_invisible` for lossy JPEG XL images should be 0.

**Screenshots**
截圖 2024-04-20 23 18 17

**Environment**
 - OS: macOS Sonoma 14.4.1 (23E224)
 - Compiler version: Apple clang version 15.0.0 (clang-1500.3.9.4)
 - CPU type: x86_64
 - cjxl/djxl version string: JPEG XL encoder v0.10.2 0.10.2 [AVX2,SSE4,SSE2]

### PR DESCRIPTION
**Pull request by zond**
### Description

Set the flag --keep_invisible to default to 1 for lossless processing, and 0 for lossy.

### Pull Request Checklist

- [x] **CLA Signed**: Have you signed the [Contributor License Agreement](https://code.google.com/legal/individual-cla-v1.0.html) (individual or corporate, as appropriate)? Only contributions from signed contributors can be accepted.
- [x] **Authors**: Have you considered adding your name to the [AUTHORS](AUTHORS) file?
- [x] **Code Style**: Have you ensured your code adheres to the project's coding style guidelines? You can use `./ci.sh lint` for automatic code formatting.

Please review the full [contributing guidelines](https://github.com/libjxl/libjxl/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: tools/cjxl_main.cc [hunk 1]
```diff
@@ -683,8 +683,16 @@ void ProcessFlags(const jxl::extras::Codec codec,
...
+  if (args->keep_invisible == jxl::Override::kDefault) {
+    int64_t value =
+        (args->distance == 0.0 || args->lossless_jpeg == JXL_TRUE) ? 1 : 0;
+    params->options.emplace_back(JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE, value,
+                                 0);
+  } else {
+    int64_t value = args->keep_invisible == jxl::Override::kOn ? 1 : 0;
+    params->options.emplace_back(JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE, value,
+                                 0);
+  }
```
Comment (jonsneyers) [inline tools/cjxl_main.cc:695]: It shouldn't be necessary to duplicate this logic in cjxl — it should already be in libjxl, if JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE gets set to -1 or if it doesn't get set at all.

If libjxl isn't behaving as expected then the bug should get fixed there.

Comment (zond) [reply tools/cjxl_main.cc]: Good point.

Changed it around, see what you think.

**Review by mo271 — APPROVED** [review]

File: tools/cjxl_main.cc (hunk 1 above)
Comment (zond) [reply tools/cjxl_main.cc]: I'm guessing @mo271's approval makes this discussion redundant?

**Review by jonsneyers — APPROVED** [review]

**Comment by zond** [comment]
@jonsneyers: I talked with @mo271 and were discussing how to actually verify that it solves the problem.

I have so far been unable to actually repro the reported error, I suspect my example PNG doesn't look like it should.

I'll  bring this up in the issue thread.

**Pull request merged**